src/openclsim/critical_path/simulation_graph.py

```diff
@@ -53,7 +98,48 @@ def __check_rec(self, recorded_activity_df):
         recorded_activity_df : pd.DataFrame
             attribute from instance of BaseCpLog
         """
-        # ..
+        # check expected columns
+        missing_columns = [
+            c
+            for c in self.__SUPERLOG_COLUMNS.keys()
+            if c not in recorded_activity_df.columns
+        ]
+        assert missing_columns == [], f"cp_log is missing columns {missing_columns}"
+
+        return self.__prepare_recorded(recorded_activity_df)
+
+    def __prepare_recorded(self, recorded_activity_df):
+        """
+        Prepare the recorded_activity_df to a workable format.
+
+        Makes sure that column names are standardized, and that durations of
+        activities are presented in seconds.
+        """
+        # rename columns to class defaults
+        recorded_activity_df = recorded_activity_df.rename(
+            columns=self.__SUPERLOG_COLUMNS
+        )
+
+        # make sure the duration is numeric
+        recorded_activity_df["duration"] = (
+            recorded_activity_df["end_time"] - recorded_activity_df["start_time"]
+        )
+        if isinstance(recorded_activity_df["duration"][0], dt.timedelta):
+            logging.debug("Converting duration to seconds (float)")
+            recorded_activity_df["duration"] = round(
+                recorded_activity_df["duration"].dt.total_seconds(), 3
+            )
+        if not isinstance(recorded_activity_df["duration"][0], (float, int)):
+            raise TypeError(
+                f"Duration computed as type {type(recorded_activity_df['duration'][0])} "
+                "is not supported!"
+            )
+
+        # make sure all durations are positive
+        assert all(
```

Comment:
nicer would be
```
if not all(***):
   raise SomeError(***)
```

But then we would need to change all coming assertions accordingly... Any idea of a standard of OpenCLSim concerning assertions?

Reply:
I believe that assertion is indeed not preferred here, changed to ValueError